### PR TITLE
Ethereal fixes

### PR DESCRIPTION
**Pull request by DeityLink**
- incorporeal_move values are now defines.
- you can no longer drop items while ethereal
- turrets no longer track/attack you while you are ethereal
- projectiles move through you and explosions don't affect you while you are ethereal

### CODE REVIEW

**Comment by clusterfack** [comment]
Dumby dont make my precious ethereal jaunt snowflake code for no fucking reason.

Check if flags & INVULNERABLE, mobs in ethereal jaunt have that flag until they are released from the jaunt.

File: code/modules/mob/inventory.dm [hunk 1]
```diff
@@ -145,6 +145,10 @@
 //Drops the item in our hand - you can specify an item and a location to drop to
 /mob/proc/drop_item(var/obj/item/to_drop, var/atom/Target)
 	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/mob/proc/drop_item() called tick#: [world.time]")
+
+	if(incorporeal_move == INCORPOREAL_ETHEREAL) //can't drop items while etheral
```
Comment (clusterfack) [inline code/modules/mob/inventory.dm]: Better solved by the introduction of a flag than this, or by whatever method we fixed dropping while ventcrawling with.

**Comment by DeityLink** [comment]
@clusterfack I'd argue that being invulnerable and being etheral are entirely different. I can add an ETHEREAL flag if that's what you want.

**Comment by clusterfack** [comment]
What do you mean you can argue? You literally get the invulnerable flag. What the fuck are you on about?

File: code/modules/mob/mob_movement.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /mob/CanPass(atom/movable/mover, turf/target, height=1.5, air_group = 0)
-	if(air_group || (height==0)) return 1
+	if(air_group || (height==0) || (incorporeal_move == INCORPOREAL_ETHEREAL)) return 1
```
Comment (clusterfack) [inline code/modules/mob/mob_movement.dm]: If this is actually a problem set density = 0 and density = 1 in jaunt, this is spaghetti

File: html/changelogs/DeityLink_5947.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+author: Deity Link
+delete-after: true
+changes:
+  - bugfix: you can no longer drop items while ethereal
+  - bugfix: turrets no longer track/attack you while you are ethereal
+  - bugfix: projectiles move through you and explosions don't affect you while you are ethereal
```
Comment (clusterfack) [inline html/changelogs/DeityLink_5947.yml]: The second of these should already be a thing

Comment (DeityLink) [inline html/changelogs/DeityLink_5947.yml]: looks like it

File: code/modules/mob/mob_movement.dm (hunk 2 above)
Comment (DeityLink) [inline code/modules/mob/mob_movement.dm]: -wizard gets knocked down, his density becomes 0
-he uses etheral jaunt, his density is still 0
-while etheral he gets up, his density becomes 1
-he gets hit by a laser while still etheral
blargharbl

**Comment by DeityLink** [comment]
I mean that ETHEREAL is another layer on top of INVULNERABLE. Also I don't know about that, since Wraith could still get killed by turrets while etheral.

**Comment by clusterfack** [comment]
That means the wraith take damage procs are shit and don't check invulnerability, or that the turrets target procs is shit and doesn't check invulnerability. We don't need a second layer when we have something that already exists that does the same thing.

File: code/modules/mob/mob_movement.dm (hunk 2 above)
Comment (clusterfack) [inline code/modules/mob/mob_movement.dm]: You cant code around that? I managed to do it just fine for tricky variables

Comment (DeityLink) [inline code/modules/mob/mob_movement.dm]: how would you code around that? I don't see it, without adding more snowflake in the "get up" code.

Comment (clusterfack) [inline code/modules/mob/mob_movement.dm]: Check if the person is lying

File: code/modules/mob/inventory.dm (hunk 1 above)
Comment (DeityLink) [inline code/modules/mob/inventory.dm]: I've replaced every other check but for now let's keep this one.

**Pull request merged**